app/routes/__index/domains/index.tsx

```diff
@@ -1,17 +1,28 @@
 import { AddIcon } from '@chakra-ui/icons';
 import { Button, Container, Flex, Heading, Text } from '@chakra-ui/react';
+import type { LoaderArgs } from '@remix-run/node';
 import { Link, useNavigate } from '@remix-run/react';
+import { typedjson, useTypedLoaderData } from 'remix-typedjson';
 import type { DomainsTableAction } from '~/components/domains-table';
 import DomainsTable from '~/components/domains-table';
-import DOMAINS_MOCK from '~/mocks/domains';
+import { getRecordsByUsername } from '~/models/record.server';
+import { requireUsername } from '~/session.server';
+import type { Record } from '@prisma/client';
+
+export const loader = async ({ request }: LoaderArgs) => {
+  const username = await requireUsername(request);
+
+  return typedjson(await getRecordsByUsername(username));
```

Comment:
So `typedjson` is sort of like `JSON.parse()` but for TypeScript?

Reply:
I would say more like `json` function from remix but with better support of types